PennMobile/Login/PathAtPennNetworkManager.swift

```diff
@@ -0,0 +1,159 @@
+//
+//  PathAtPennNetworkManager.swift
+//  PennMobile
+//
+//  Created by Anthony Li on 9/25/22.
+//  Copyright © 2022 PennLabs. All rights reserved.
+//
+
+import Foundation
+import SwiftSoup
+
+enum PathAtPennError: Error {
+    /// The data was a malformed string.
+    case corruptString
+
+    /// A token was not found in the Path@Penn authorization response.
+    case noTokenFound
+
+    /// The returned ``URLResponse`` was not an ``HTTPURLResponse``.
+    case notHttpResponse
+
+    /// The API returned an unexpected status code.
+    case unexpectedStatus(Int)
+
+    /// The student data was not returned from Path@Penn.
+    case noStudentData
+}
+
+class PathAtPennNetworkManager {
+    static let instance = PathAtPennNetworkManager()
+}
+
+// MARK: - Path@Penn Authentication
+
+extension PathAtPennNetworkManager {
+    static private let oauthURL = URL(string: "https://idp.pennkey.upenn.edu/idp/profile/oidc/authorize?response_type=code&scope=openid+email+profile&client_id=courses.upenn.edu%2Fsam_rMReby8Vl3M1BiyVWMAT&redirect_uri=https%3A%2F%2Fcourses.upenn.edu%2Fsam%2Fcodetotoken")!
+
+    /// Fetches and returns a Path@Penn auth token.
+    func getToken() async throws -> String {
+        let (data, _) = try await URLSession.shared.data(from: PathAtPennNetworkManager.oauthURL)
+        let str = try String(data: data, encoding: .utf8).unwrap(orThrow: PathAtPennError.corruptString)
+        let matches = str.getMatches(for: "value: \"(.*)\"")
+        if matches.isEmpty {
+            print(str)
+        }
```

Comment:
maybe remove this print statement

Reply:
Oh yeah, I should remove that. That was originally there for debugging cases where Path@Penn wouldn't return a token, but we'll need to resolve that later with PennLoginController, probably